src/backend/mpi/swdsm.cpp

```diff
@@ -1031,7 +1031,13 @@ void argo_initialize(std::size_t argo_size, std::size_t cache_size){
 		pthread_barrier_init(&threadbarrier[i],NULL,i);
 	}
 
-	cachesize = cache_size+pagesize*CACHELINE;
+	cachesize = 0;
+	if(cache_size > argo_size) {
+		cachesize += argo_size;
```

Comment:
The problem here is with requesting a total Argo memory size smaller than the default cache size. If there is no automatic resizing, then this usage becomes invalid, or unnecessarily slow. Right now, if a cache is larger than the total memory size, then the exceeding cache size will simply be unused.

Also, the += is for consistency with the lines following these. Technically not needed, but I wanted to express "sum of these components" in the code.

Reply:
If someone asks for a total Argo memory smaller than the cache size, shouldn't we give them an error? Or at least a warning message. Because it sounds like a mistake.

Reply:
when relying on defaults it does not sound like a mistake to me. When manually specified it might be an error, or it might be the result of some scripts setting weird values. I don't think we have to hand-hold users here, but if the problem actually occurs I would add such a check.

Reply:
And what exactly prevents us to document this? (perhaps as part of this PR?)

Reply:
Nothing prevents us from documenting this, but it is not obvious to me where to document something like this. Note that it is functionally not different from creating a cache as large as requested (as the larger cache would simply go unused). This is mainly a performance patch that speeds up testing significantly.

Reply:
See issue #25 for this topic.